### PR TITLE
Add password visibility toggle to all password inputs

### PR DESCRIPTION
**Pull request by louischan-oursky**
ref #1942

### CODE REVIEW

File: resources/authgear/templates/en/web/enter_password.html [hunk 1]
```diff
@@ -21,17 +21,11 @@ <h1 class="primary-txt text-center margin-0 text-xl font-bold">
...
-	autocomplete="current-password"
-	autocapitalize="none"
-	name="x_password"
-	placeholder="{{ template "password-placeholder" }}"
->
+{{ template "__password_input.html" (dict
+	"Name" "x_password"
+	"Type" "old-password"
+	"AutoFocus" true
+) }}
```
Comment (carmenlau) [inline resources/authgear/templates/en/web/enter_password.html:28]: We should also update the new parameter `Type` in `create_password.html` and `reset_password.html`.

Comment (louischan-oursky) [reply resources/authgear/templates/en/web/enter_password.html:28]: Fixed.

File: authui/src/password-policy.ts [hunk 2]
```diff
@@ -1,106 +1,151 @@
...
+    checkPasswordDigit(value, this.digitTarget);
+    checkPasswordSymbol(value, this.symbolTarget);
+    checkPasswordStrength(
+      value,
+      this.currentMeterTarget,
+      this.currentMeterDescriptionTarget,
+      this.requiredMeterTarget,
+      this.strengthTarget
+    );
+  }
```
Comment (carmenlau) [inline authui/src/password-policy.ts:150]: Got following errors when some of the password policy is not enabled.
```
authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1 Error invoking action "input->password-policy#check"

Error: Missing target element "requiredMeter" for "password-policy" controller
    at a.get (authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1:44553)
    at a.value (authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1:1011089)
    at e.value (authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1:7029)
    at e.value (authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1:6587)
    at e.value (authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1:1972)
```

Seems we should check the existence of the target before accessing by `this.has[Name]Target`

https://stimulus.hotwired.dev/reference/targets

File: authui/src/password-policy.ts [hunk 3]
```diff
@@ -1,106 +1,151 @@
...
+    checkPasswordLowercase(value, this.lowercaseTarget);
+    checkPasswordDigit(value, this.digitTarget);
+    checkPasswordSymbol(value, this.symbolTarget);
+    checkPasswordStrength(
+      value,
+      this.currentMeterTarget,
+      this.currentMeterDescriptionTarget,
+      this.requiredMeterTarget,
+      this.strengthTarget
+    );
```
Comment (carmenlau) [inline authui/src/password-policy.ts:149]: Got following errors when some of the password policy is not enabled.
```
authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1 Error invoking action "input->password-policy#check"

Error: Missing target element "uppercase" for "password-policy" controller
    at a.get (authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1:44553)
    at a.value (authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1:1010377)
    at e.value (authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1:7029)
    at e.value (authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1:6587)
    at e.value (authgear.b2fd14d489b8a2da7358cfe4c8a0e9c0.js:1:1972)
```

Seems we should check the existence of the target before accessing by `this.has[Name]Target`

https://stimulus.hotwired.dev/reference/targets

**Pull request merged**